### DISCUSSION TITLE
portable-ruby is using EOL Ruby version (2.6.8)

### DISCUSSION THREAD

**Discussion by coaxial** [Homebrew/discussions#4367] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: cf6614b50eac1c9f4bf386dc6989f0d443148df8
Last commit: 3 days ago
Core tap JSON: 30 Mar 12:36 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_EDITOR: /usr/bin/nvim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit skylake
Clang: N/A
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /bin/curl
Kernel: Linux 5.15.0-67-generic x86_64 GNU/Linux
OS: Linux Mint 20.3 (una)
Host glibc: 2.31
/usr/bin/gcc: 9.4.0
/usr/bin/ruby: 2.7.0
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "bin" was not found in your PATH.
Consider setting your PATH for example like so:
  echo 'export PATH="/home/linuxbrew/.linuxbrew/bin:$PATH"' >> ~/.profile
```

### Description of issue

When installing brew using `NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`, I could see it pulls portable-ruby as the Ruby runtime. I could also see this:
```
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:fc45ee6eddf4c7a17f4373dde7b1bc8a58255ea61e6847d3bf895225b28d072a
######################################################################## 100.0%
==> Pouring portable-ruby-2.6.8_1.x86_64_linux.bottle.tar.gz
```
[Ruby 2.6 is EOL since 2022-04-12](https://www.ruby-lang.org/en/downloads/branches/) and the latest 2.6 is [2.6.10](https://www.ruby-lang.org/en/news/2022/04/12/ruby-2-6-10-released/). Would it make sense to use a more recent Ruby then? 2.7 is EOL tomorrow but 3 is supported for a while.

**Comment by SMillerDev** [comment]
Unfortunately we stay in sync with the ruby on macOS to avoid having to develop compatibility with a bunch of ruby versions.

**Reply by MikeMcQuaid** [reply]
Came here to say the same thing but: I checked and this is 2.6.10 now so, yeh, we should update.

**Reply by SMillerDev** [reply]
That's just as EOL as 2.6.8 though.

**Reply by coaxial** [reply]
It is but it fixes 3 7.5 CVEs so it's slightly better I guess.
- https://nvd.nist.gov/vuln/detail/CVE-2021-41817
- https://nvd.nist.gov/vuln/detail/CVE-2021-41819
- https://nvd.nist.gov/vuln/detail/CVE-2022-28739

**Reply by carlocab** [reply]
> That's just as EOL as 2.6.8 though.

@SMillerDev would [deprecate and disable](https://docs.brew.sh/Deprecating-Disabling-and-Removing-Formulae) macOS Ruby if they could

**Comment by coaxial** [comment]
MacOS ships with EOL Ruby? I don't mean this as a snarky comment, I'm curious and don't have any Mac to check.

**Reply by carlocab** [reply]
Some (currently maintained) versions do. The latest version ships with 2.6.10, so we could/should probably update `portable-ruby` to that.

Apple hopes to be able to get rid of the built-in `ruby`, so I guess keeping it up-to-date isn't high on their list of priorities.

**Comment by coaxial** [comment]
I'm confused though. Is this portable-ruby version only used to run `brew` itself or is it also what's used for formulas that need Ruby to run? Can formulas that need Ruby to run pack their own (up to date) runtime or do they all run on the 2.6.8 that `brew` sets up?

**Reply by SMillerDev** [reply]
It is only used to run `brew` itself. Formula depend on other formula (or system components) but `portable-ruby` is neither.
